Add data warehouse glossary (#1652)
fishtown-analytics/docs.getdbt.com: website/docs/terms/data-warehouse.md

Not to mention, you will save your analytics engineer and data analyst a few headaches. They would otherwise have to manually pull needed data from various sources. Not having a single source of truth decreases your data quality, wastes your data team’s precious time, and makes it difficult to combine data from different sources. 

### Ability to control who has access and the type of access they have

Data warehouses have extensive security features which allow you to control who has access to what. You have the ability to give someone as little or extensive permissions as you wish. Warehouses give you the ability to create users and assign them roles. Each role has its own set of permissions to which databases and <Term id="table">tables</Term> it can see. Then, you can also choose who is allowed to query those tables or even update and delete them. 

> When anyone in your org can easily access your data, bad things can happen. You riskthe potential of important data being deleted or incorrectly edited. If you are using Excel spreadsheets for data pulls, this is especially prone to human error. 

Reviewer: ```suggestion
When anyone in your organization can easily access your data, bad things can happen. You risk the potential of important data being deleted, incorrectly edited, or inappropriately accessed. Data warehouses users, roles, policies, and security measures can help ensure data is in the right hands of the right people.
```

### Fast reporting

> Because all of your data is located in the same place, it allows for faster reporting compared to pulling data from many different sources. You don’t need to make a bunch of data pulls, upload them to a central location, and then somehow join the datasets. Not to mention the slow speed at which that occurs. A central location allows for you to quickly access and query millions of rows of data, allowing transformations and reporting to be done much faster. 

Reviewer: ```suggestion
Because all of your data is located in the same place, it allows for faster reporting compared to pulling data from many different sources. A central location allows for you to quickly access and query millions of rows of data, allowing transformations and reporting to be done much faster. 
```

## Data platforms that support data warehousing workloads

| **Data platform** | **Description** | 
|---|---|
| Snowflake | Snowflake is a fully-managed platform for data warehousing, data lakes, data engineering, data science, and data application development. |
| Databricks | Databricks is a cloud-based collaborative data science, data engineering, and data analytics platform that combines the best of data warehouses and data lakes into a lakehouse architecture. |
| Google BigQuery | Google BigQuery is a serverless, highly scalable data warehouse that comes with a built-in query engine. |
| Amazon Redshift | Amazon Redshift is a fully-managed petabyte-scale cloud-based data warehouse designed for large scale data set storage and analysis. |
| Postgres | PostgreSQL is an advanced, enterprise class open source relational database that supports both SQL (relational) and JSON (non-relational) querying. |

## Data warehouse vs data lake

> A data lake is a system where you store, process, and query unstructured, semi-structured, and structured data at almost any scale. The main difference between a data warehouse and a data lake is the type and way data is stored. Data warehouses contain structured data that is meant to organize data for analytics use. Data lakes can contain pretty much any kind of data--structured or unstructured--and data is usually left it its raw format until it's ready to use. Compare that to data warehouses, whose primary goal is to be a place for data teams to store both raw and transformed, usable data.

Reviewer: ```suggestion
A data lake is a system where you store, process, and query unstructured, semi-structured, and structured data at almost any scale. The main difference between a data warehouse and a data lake is the type and way data is stored. Data warehouses contain structured data that is meant to organize data for analytics use. Data lakes can contain pretty much any kind of data—structured or unstructured—and data is usually left it its raw format until it's ready to use. Compare that to data warehouses, whose primary goal is to be a place for data teams to store both raw and transformed, usable data.
```

> It is easy to confuse a data warehouse for a data lake. Unlike a data warehouse, a data lake is a system where you store, process, and query unstructured, semi-structured, and structured data at any scale. The main difference is between the type of data that is stored. Data warehouses contain structured data such as integers, strings, and timestamps to name a few. Data lakes can contain any kind of data- structured or unstructured. Unstructured data can include text files, audio files, video files, and images. Data warehouses store data to be used for analytics purposes. If it can’t be used for analysis, it won’t be stored in a warehouse. Data lakes store all data, whether it is used for analytics or not. 

Reviewer: ```suggestion
A data lake is a system where you store, process, and query unstructured, semi-structured, and structured data at almost any scale. The main difference between a data warehouse and a data lake is the type and way data is stored. Data warehouses contain structured data that is meant to organize data for analytics use. Data lakes can contain pretty much any kind of data--structured or unstructured--and data is usually left it its raw format until it's ready to use. Compare that to data warehouses, whose primary goal is to be a place for data teams to store both raw and transformed, usable data.
```

## Conclusion

Data warehouses have come a long way [in the last 40 years](https://www.getdbt.com/blog/future-of-the-modern-data-stack/). They began as a physical location with huge costs associated with them to a system available to anyone, anywhere at an affordable cost. They have the power to centralize all of your business’s data, allowing for faster analytics operations, standardized KPIs, and a single source of truth. All businesses need a data warehouse in order to operate quickly and efficiently with data that they can rely on. The question isn’t whether or not you need a data warehouse, but which data warehouse you should choose. Make a list of the key features needed for your business and use that to assess the options above. 

## Additional reading

> - [Operational analytics](https://www.notion.so/dbtlabs/Glossary-Data-warehouse-b6ac46004f3a4813bc34e33efeb4c773#6e658bc658f44036979c4d96b46620f3)

Reviewer: We should probably remove this link or link to something else? This is a Notion page hahah.

Author: Ah, good catch. Must have accidentally copied the block link instead of the hyperlink. Thanks!

- [Glossary: ETL](https://docs.getdbt.com/terms/etl/)

> - [Glossary: ELT](https://docs.getdbt.com/terms/elt/)]

Reviewer: ```suggestion
- [Glossary: ELT](https://docs.getdbt.com/terms/elt/)
```
